src/EhrgoHealth.WebService/Models/AllergyIntolerance.cs

```diff
@@ -0,0 +1,164 @@
+        /// </summary>       
+        /// <param name="patientID"> The patient's ID from FHIR</param>
+        /// <param name="medications"> The List of medications they currently know their patient is taking</param>
+        public Boolean IsAllergicToMedications(uint patientID, List<String> medications)
+        {
+            //First let us fetch the known allergies of the patient.
+            List<string> returnedAllergies = new List<string>();
+            Dictionary<string, Boolean> lookupPatientsKnownAllergies = GetPatientsKnownAllergies(patientID);
+            if (lookupPatientsKnownAllergies == null)
+            {
+                //There are no records on the FHIR server of this patient having any allergies.
+                return false;
+            }
+            return IsAllergic(medications, lookupPatientsKnownAllergies);
+
+        }//end IsAllergicToMedications method
+
+
+
+        /// Private helper method for comparing the patient's medications against their known allergies
+        /// <summary>       
+        /// I'm the least happy with this method, because I know I can refactor this to do more to also give me a list or dictionary of my results.
+        /// </summary>       
+        /// <param name="medications"> List of patients medications</param>
+        /// <param name="lookupPatientsKnownAllergies"> Dictonary of patient's known allergies</param>
+        private Boolean IsAllergic(List<string> medications, Dictionary<string, Boolean> lookupPatientsKnownAllergies)
+        {
+            List<string> currentAllergyCodeList = new List<string>();
+            foreach (var m in medications)
+            {
+                allergyLookup.TryGetValue(m.ToLower(), out currentAllergyCodeList);
+                if (currentAllergyCodeList != null && currentAllergyCodeList.Count > 0)
+                {
+                    foreach (var c in currentAllergyCodeList)
+                    {                        
+                        if (lookupPatientsKnownAllergies.ContainsKey(c))
+                        {                           
+                            return true;
+                        }
+                    }
+                }
+
+            }
+            return false;
+        }//end IsAllergic method
+
+        /// Returns a list of patient's allergies from the FHIR server
+        /// <summary>       
+        /// If you want the list of allergy codes of a patient, then just pass in the patient's ID.
+        /// </summary>       
+        /// <param name="medications"> List of patients medications</param>
+        /// <param name="lookupPatientsKnownAllergies"> Dictonary of patient's known allergies</param>
+        public Dictionary<string, Boolean> GetPatientsKnownAllergies(uint patientID)
+        {
+            var allergyResource = fhirClient.Read<Hl7.Fhir.Model.AllergyIntolerance>("AllergyIntolerance/" + patientID);

```

Comment:
you have a using Hl7.Fhir.Model statement at the top of this file you shouldn't need this here...also refactoring essentials plugin for VS would have grayed this text out.